### PR TITLE
Refactor isolate test

### PR DESCRIPTION
**Pull request by dnfield**
Split from #16824 

This is intended to make it easier for other test harnesses to launch a Dart VM.  See linked PR for example of using it.

### CODE REVIEW

**Comment by dnfield** [comment]
Looks like the mac test is going to timeout.  /cc @yjbanov  - the felt test on mac for this run has already taken > 51 minutes and appears to not be making progress.

I'm going to ignore the test for this change since the change cannot possibly be impacting web code, it's a C++ only test based change.

@yjbanov @nturgut - maybe we could have the web tests skip running if the only changes are to *.cc|*.h|*.mm|*.m|*.java files?

File: testing/dart_isolate_runner.h [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  }
+
+ private:
+  std::shared_ptr<DartIsolate> isolate_;
+  fml::RefPtr<fml::TaskRunner> runner_;
+
+  FML_DISALLOW_COPY_AND_ASSIGN(AutoIsolateShutdown);
+};
+
+static void RunDartCodeInIsolate(DartVMRef& vm_ref,
```
Comment (chinmaygarde) [inline testing/dart_isolate_runner.h]: As soon as this header gets included in more than one translation unit, there are going to be duplicate symbol errors. The implementation needs to be in its own TU.

File: testing/dart_isolate_runner.h [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    return;
+  }
+
+  root_isolate->get()->AddIsolateShutdownCallback(
+      settings.root_isolate_shutdown_callback);
+
+  result = std::move(root_isolate);
+}
+
+static std::unique_ptr<AutoIsolateShutdown> RunDartCodeInIsolate(
```
Comment (chinmaygarde) [inline testing/dart_isolate_runner.h]: Ditto about needing to be in its own TU.

File: testing/dart_isolate_runner.h [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+#include "flutter/fml/thread.h"
+
+namespace flutter {
+namespace testing {
+
+class AutoIsolateShutdown {
+ public:
+  AutoIsolateShutdown() = default;
+
+  AutoIsolateShutdown(std::shared_ptr<DartIsolate> isolate,
```
Comment (chinmaygarde) [inline testing/dart_isolate_runner.h]: Now that this is a general purpose utility, please OOL these like in the rest of the engine codebase.

**Comment by chinmaygarde** [comment]
Sorry about the delay in the review. Can we please fix the OOL-ing of the static methods in a follow up patch?

**Comment by dnfield** [comment]
I'll follow up on this.

File: testing/dart_isolate_runner.h (hunk 3 above)
Comment (dnfield) [reply testing/dart_isolate_runner.h]: Fixed these in #16941

**Pull request merged**